src/components/views/rooms/BasicMessageComposer.tsx

```diff
@@ -47,7 +47,7 @@ import AutocompleteWrapperModel from "../../../editor/autocomplete";
 import DocumentPosition from "../../../editor/position";
 import {ICompletion} from "../../../autocomplete/Autocompleter";
 
-const REGEX_EMOTICON_WHITESPACE = new RegExp('(?:^|\\s)(' + EMOTICON_REGEX.source + ')\\s$');
+const REGEX_EMOTICON_WHITESPACE = new RegExp('(?:^|\\s|(?<=^\\+))(' + EMOTICON_REGEX.source + ')\\s$');
```

Comment:
this regexp is getting quite impenetrable now :) can we comment it?

`// match emoticons which follow the start of a line, whitespace, a plus at the start of a line.`

I'm a bit unclear on why we need the nested positive look-behind assertion here - plus it looks like the `(?<=)` will act as a capturing group, which might have undesirable side-effects given the wrapping group is a non-capturing `(?:)`.  Does `(?:^|\\s|^\\+)` work?

Reply:
Good point! I've gone ahead and commented it as well as removed the lookbehind and it is still working as intended.

I think the negative lookbehind is still necessary for the EmojiProvider regex. Otherwise, the entire match gets replaced when autocompleting an emoji suggestion ("+:gleeful<tab>" gets sent to "😃" instead of "+😃"). Unless there is a different way to do it.

Reply:
thanks for this (and thanks for the PR!). I guess a similar clarification is needed for `EMOJI_REGEX` then to comment why the lookbehind is needed there but not here?

otherwise this lgtm, but i'll defer to @jryans & co for formal review from the eleweb team.